pkg/funky/server.go

```diff
@@ -21,7 +21,7 @@ import (
 // Server an interface for managing function servers
 type Server interface {
 	GetPort() uint16
-	Invoke(input Message) (io.ReadCloser, error)
+	Invoke(input *Message) (io.ReadCloser, error)
```

Comment:
It's better to return `interface{}` here (with the parsed content from the `io.ReadCloser`): 
- closing the stream should be as close to the stream origin
- parsing the content of the stream may result in another error

It kinda looks like a "leaked" responsibilities, that could be handled at a lower level and not burden the router, who's Delegate() method is doing a bit too much.